models/eprop_iaf.cpp

```diff
@@ -378,33 +378,32 @@ eprop_iaf::handle( DataLoggingRequest& e )
 
 void
 eprop_iaf::compute_gradient( const long t_spike,
-  const long t_previous_spike,
-  double& previous_z_buffer,
+  const long t_spike_previous,
+  double& z_previous_buffer,
   double& z_bar,
   double& e_bar,
   double& epsilon,
   double& weight,
   const CommonSynapseProperties& cp,
   WeightOptimizer* optimizer )
 {
-  long t = t_previous_spike; // inter-spike time step
-  double e = 0.0;            // eligibility trace
-  double z = 0.0;            // spiking variable
-  double psi = 0.0;          // surrogate gradient
-  double L = 0.0;            // learning signal
-  double grad = 0.0;         // gradient
+  double e = 0.0;                // eligibility trace
+  double z = 0.0;                // spiking variable
+  double z_current_buffer = 1.0; // buffer containing the spike that triggered the current integration
+  double psi = 0.0;              // surrogate gradient
+  double L = 0.0;                // learning signal
+  double grad = 0.0;             // gradient
 
   const EpropSynapseCommonProperties& ecp = static_cast< const EpropSynapseCommonProperties& >( cp );
 
-  auto eprop_hist_it = get_eprop_history( t_previous_spike - 1 );
+  auto eprop_hist_it = get_eprop_history( t_spike_previous - 1 );
 
-  double pre = 1.0;
-
-  while ( t < std::min( t_spike, t_previous_spike + P_.eprop_isi_trace_cutoff_ ) )
+  for ( long t = t_spike_previous; t < std::min( t_spike_previous + P_.eprop_isi_trace_cutoff_, t_spike );
```

Comment:
I think that using a function call (`std::min`) directly in the loop condition can make it harder to quickly understand the bounds of the loop. Maybe an alternative could be:

```cpp
const long t_spike_limit = std::min(t_spike_previous + P_.eprop_isi_trace_cutoff_, t_spike);

for (long t = t_spike_previous; t < t_spike_limit; ++t, ++eprop_hist_it)
```

Reply:
Good suggestion! While `t_spike_limit` already goes in the right direction, it might convey that there is a spike at `t_spike_previous + cutoff`. For example, `t_computation_limit` / `t_compute_limit` or `t_compute_until` would circumvent that problem.
```
const long t_compute_until = std::min(t_spike_previous + P_.eprop_isi_trace_cutoff_, t_spike);

for (long t = t_spike_previous; t < t_compute_until; ++t, ++eprop_hist_it)
```

Reply:
That's true.

Yeah I think `t_compute_until` is good.